compiler/src/org.graalvm.compiler.asm.amd64/src/org/graalvm/compiler/asm/amd64/AMD64Assembler.java

```diff
@@ -2512,6 +2597,26 @@ void pcmpestri(Register dst, Register src, int imm8) {
         emitByte(imm8);
     }
 
+    public final void pmovzxbw(Register dst, AMD64Address src) {
+        assert supports(CPUFeature.SSE4_2);
+        // XXX legacy_mode should be: _legacy_mode_bw
```

Comment:
This is the only thing I'm worried about.  Not sure what `_legacy_mode_bw` really is.

Reply:
It looks like part of the magic autoencoding for supporting AVX512.  We don't really support AVX512 right now I think.  Maybe @rschatz has some thoughts on how we should deal with?  I suspect we leave it as false but maybe we also need to explicitly disable the AVX512 feature tests?

Reply:
I can only speculate, I don't really know what that means either.

It seems the `legacy_mode` attribute is never actually used in our code.
From the name, I suspect it has something to do with selecting EVEX vs VEX prefix. Inside the constructor of `AMD64InstructionAttr` the `legacy_mode` flag is filtered by the `AVX512F` feature flag, which tells us whether `EVEX` is available. And `_legacy_mode_bw` sounds like it has something to do with the `AVX512BW` feature flag. The `VPMOVZXBW` needs the `AVX512BW` feature flag to be usable with `EVEX`.